cpp/include/copc-lib/io/laz_base_writer.hpp

```diff
@@ -0,0 +1,61 @@
+#ifndef COPCLIB_IO_LAZ_BASE_WRITER_H_
+#define COPCLIB_IO_LAZ_BASE_WRITER_H_
+
+#include <array>
+#include <iterator>
+#include <optional>
+#include <ostream>
+#include <stdexcept>
+#include <string>
+
+#include "copc-lib/copc/copc_config.hpp"
+#include "copc-lib/geometry/vector3.hpp"
+#include "copc-lib/las/header.hpp"
+#include "copc-lib/las/laz_config.hpp"
+#include "copc-lib/las/points.hpp"
+#include "copc-lib/las/utils.hpp"
+#include "copc-lib/las/vlr.hpp"
+
+namespace copc::laz
+{
+
+class BaseWriter
+{
+  public:
+    BaseWriter(std::ostream &out_stream, std::shared_ptr<las::LazConfigWriter> laz_config_writer)
+        : out_stream_(out_stream), config_(laz_config_writer)
+    {
+        // reserve enough space for the header & VLRs in the file
+        std::fill_n(std::ostream_iterator<char>(out_stream_), FirstChunkOffset(), 0);
+        open_ = true;
+    }
+
+    virtual void Close();
+    // Call close on destructor if needed
+    ~BaseWriter() { Close(); }
+
+  protected:
+    const uint32_t VARIABLE_CHUNK_SIZE = (std::numeric_limits<uint32_t>::max)();
+
+    // 8 bytes for the chunk table offset
+    uint64_t FirstChunkOffset() const { return OffsetToPointData() + sizeof(uint64_t); };
+
+    // Base constructor used in WriterInternal Class
+    explicit BaseWriter(std::ostream &out_stream) : out_stream_(out_stream), config_(nullptr) {}
```

Comment:
Why set config here? Should it just remain unset?

Reply:
Tru dat